### ISSUE TITLE
Improvements to Validation Mechanism & Translation Failure Logic

### ISSUE THREAD

**Issue by cb-steven-matison** [moryakovdv/iwxxmConverter#12]
I am creating this issue as a holder for some related Recommendations for Improvement.  I would like this to start discussions and drive future improvements to the iwxxmConverter Repository.

1. Adjust Validation Mechanism 
2. Pass Translation Failures to JSON Response translationErrors
3. Attempt to translate all Parsed objects during Conversion

With the current validation method it is possible to successfully validate (true) an empty XML Object against Schematron Rules.  Although this is logically valid, for various different test case scenarios the actual TAC message is invalid and fails to Translate.  It would make sense to adjust this behavior and/or not validate the empty XML. 

Additionally, in order to prepare this system to be more robust in production environments, I think we should work to pass Translation Failure exceptions to a new JSON Object within the API response.     This behavior would then clearly indicate to the End Users all of the exception points where a given translation fails.  Currently the behavior is a hard stop when the first object fails.  Subsequent objects are not checked in these scenarios.

**Comment by cb-steven-matison** [comment]
Just some updates on my progress here.   I have created a wrapper API that uses the converter and validator.   I have been through the entire Validator.  I have a made an SValidator which validates the generated XML against the iwxxm schema (not Schematron).   My results have been pretty interesting but I am able to pass the conflicts to asserts and get the response to return valid:false.   The xml testing here should drive some more simple source chances in the converter to satisfy missing elements.  Here is one example:

```
Attribute 'uom' must appear on element 'iwxxm:meanWindSpeed'."
```

This also re-affirms we need to discuss where the current converter needs to create empty objects as well as what really is or isn't required.  Thats a big topic for validation and translations.

**Comment by steven-matison** [comment]
Back on this topic.  While thinking about validation in the last 2 weeks I wondered how to see/test validation:false specific errors with a schematron conflict.     I am pretty sure I seen the code in place to do so but I have never seen it in action (no test cases have returned schematron failure in console or any error json object).    There was also similar but different mechanism in V2  so I wonder if there are issues still in V3?  It could even be something in my project build?   I think we need a unit Test to test some bad xml against schematron since all the converted XML passes (even an empty iwxxm xml object passes).

When I built my SValidator I had issues with the existing validators Assertion list ("key" : "value"), and ended up just using a regular list ("value").   When I passed errors into this list I could see valid:false as well as debug the failures into console.  I could never debug schematron validation so I suspect maybe something isn't working right.
